packages/aws-cdk-lib/aws-logs/lib/subscription-filter.ts

```diff
@@ -57,6 +58,10 @@ export class SubscriptionFilter extends Resource {
       physicalName: props.filterName,
     });
 
+    if (props.distribution && !(props.destination instanceof KinesisDestination)) {
```

Comment:
```suggestion
    if (
      props.distribution &&
      !Token.isUnresolved(props.distribution) && 
      !Token.isUnresolved(props.destination) && 
      !(props.destination instanceof KinesisDestination)
    ) {
```
We should check for unresolved values on validation.